addons/sale_coupon/__manifest__.py

```diff
@@ -6,20 +6,14 @@
     'description': """Integrate coupon mechanism in sales orders.""",
     'category': 'Sales/Sales',
     'version': '1.0',
-    'depends': ['sale_management'],
+    'depends': ['coupon', 'sale_management'],
     'data': [
-        'security/ir.model.access.csv',
         'wizard/sale_coupon_apply_code_views.xml',
-        'wizard/sale_coupon_generate_views.xml',
         'views/sale_order_views.xml',
-        'views/sale_coupon_views.xml',
-        'views/sale_coupon_program_views.xml',
+        'views/coupon_views.xml',
+        'views/coupon_program_views.xml',
         'views/res_config_settings_views.xml',
-        'report/sale_coupon_report.xml',
-        'report/sale_coupon_report_templates.xml',
-        'data/sale_coupon_email_data.xml',
     ],
     'demo': [
```

Comment:
Nitpicking: if there's no demo anymore, remove the `demo` key from the manifest file